x/programs/runtime/dependencies.go

```diff
@@ -22,9 +24,14 @@ type Runtime interface {
 	WriteGuestBuffer(context.Context, []byte) (uint64, error)
 	// Stop performs a shutdown of the engine.
 	Stop(context.Context) error
+	// Gets the user defined functions
+	GetUserData(ctx context.Context) (map[string]int, error)
+
+   GetCurrentGas(ctx context.Context) uint64
```

Comment:
these methods are used for the server in order grab info about the runtime. I don't necessarily think we should require these in the interface tho...

also need to fmt this